src/main/java/mate/academy/service/UserService.java

```diff
@@ -15,6 +15,12 @@ public class UserService {
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
+
+        for (int i = 0; i < users.length; i++) {
+            if (users[i].getEmail().equals(email)) {
+                return users[i];
+            }
+        }
```

Comment:
You can do like Map userEmailToPassword, to avoid O(n)

Reply:
I am confident that using a map can effectively optimize the code and speed up its compilation. However, I have not yet studied this in this course, so I cannot apply it. Most likely, this will be explained in the next topics, so do not consider my solution incorrect. I have removed the unnecessary lines you pointed out, as can be seen in the following commit